### PR TITLE
Chapter18 regions

### PR DESCRIPTION
**Pull request by HannaMichaelis**
## Summary

Contributes to Intellitect/EssentialCSharpManuscript#116

### CODE REVIEW

File: src/Chapter18/Listing18.18.UpdatingCommandLineHandler.TryParseToHandleAliases.cs [hunk 1]
```diff
@@ -76,7 +82,10 @@ public class CommandLineHandler
                         commandLine, optionParts[1], null);
                     success = true;
                 }
-                else if(property.PropertyType.GetTypeInfo().IsEnum)
+                else if(
+                    //property.PropertyType.IsEnum also available
```
Comment (BenjaminMichaelis) [inline src/Chapter18/Listing18.18.UpdatingCommandLineHandler.TryParseToHandleAliases.cs]:
```suggestion
                    // property.PropertyType.IsEnum also available
```
nit: space after //

File: src/Chapter18/Listing18.18.UpdatingCommandLineHandler.TryParseToHandleAliases.cs [hunk 2]
```diff
@@ -85,16 +94,16 @@ public class CommandLineHandler
...
+                        $@"Data type '{ property.PropertyType.ToString() }' on {
+                        commandLine.GetType().ToString() } is not supported.";
                 }
             }
             return success;
         }
+        #endregion INCLUDE
 
-// Justification: Not fully implemented.
+        // Justification: Not fully implemented.
```
Comment (BenjaminMichaelis) [inline src/Chapter18/Listing18.18.UpdatingCommandLineHandler.TryParseToHandleAliases.cs]:
```suggestion
// Justification: Not fully implemented.
```
nit: align the comment with the #pragma warning disable

File: src/Chapter18/Listing18.20.AttributeUsageAttributeRestrictingWhereToApplyAnAttribute.cs [hunk 3]
```diff
@@ -1,8 +1,13 @@
 namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter18.Listing18_20
 {
+    //will not compile if uncommented
```
Comment (BenjaminMichaelis) [inline src/Chapter18/Listing18.20.AttributeUsageAttributeRestrictingWhereToApplyAnAttribute.cs]:
```suggestion
    // Will not compile if uncommented
```
nit: styling of comment

File: src/Chapter18/Listing18.26.DynamicProgrammingUsingReflection.cs [hunk 4]
```diff
@@ -1,18 +1,23 @@
...
         public static void Main()
         {
-            // ...
+            #endregion EXCLUDE
             dynamic data =
               "Hello!  My name is Inigo Montoya";
             Console.WriteLine(data);
             data = (double)data.Length;
             data = data * 3.5 + 28.6;
+            //
```
Comment (BenjaminMichaelis) [inline src/Chapter18/Listing18.26.DynamicProgrammingUsingReflection.cs]:
```suggestion

```
Not sure what these // are for? Can it be removed or should it be made into // ...

File: src/Chapter18/Listing18.26.DynamicProgrammingUsingReflection.cs [hunk 5]
```diff
@@ -1,18 +1,23 @@
...
             if(data == 2.4 + 112 + 26.2)
+                //The distance (in miles) for the swim, bike, and
+                //run portions of an Ironman traithlon, respectively
```
Comment (BenjaminMichaelis) [inline src/Chapter18/Listing18.26.DynamicProgrammingUsingReflection.cs]:
```suggestion
                // The distance (in miles) for the swim, bike, and
                // run portions of an Ironman traithlon, respectively
```
nit: space after //

File: src/Chapter18/Listing18.30.OverridableMembersOnSystem.Dynamic.DynamicObject.cs [hunk 6]
```diff
@@ -39,5 +40,6 @@ public class DynamicObject : IDynamicMetaObjectProvider
         public virtual bool TryUnaryOperation(
             UnaryOperationBinder binder, out object result);
     }
+    #endregion INCLUDE
```
Comment (BenjaminMichaelis) [inline src/Chapter18/Listing18.30.OverridableMembersOnSystem.Dynamic.DynamicObject.cs]:
```suggestion
```
nit: extra whitespace

**Review by BenjaminMichaelis — APPROVED** [review]
Nice job @HannaMichaelis! looks good

**Review by Keboo — APPROVED** [review]

**Pull request merged**